src/sageworks/utils/extract_model_artifact.py

```diff
@@ -84,21 +85,34 @@ def download_and_extract_model(model_artifact_uri):
             model_files = glob.glob(os.path.join(tmpdir, "*.joblib"))
             if not model_files:
                 raise FileNotFoundError("No .joblib file found in the extracted model artifact.")
-            model_file_path = model_files[0]
+            
+            # Instantiate return model
+            model_return = None
 
-            # Load the model
-            model = joblib.load(model_file_path)
+            # Check each model_file for an XGBModel object
+            for model_file in model_files:
+
+                # Load the model
+                model_object = joblib.load(model_file)
+
+                # Check type
+                if isinstance(model_object, XGBModel):
+                    print(f"{model_file} is a model object.")
+
+                    # Set return if type check passes
+                    model_return = model_object
+                else:
+                    print(f"{model_file} is NOT a model object.")
 
```

Comment:
Trying to understand the new logic here, admittedly the old logic isn't great either :)

Looks like it loops around and basically picks the ^last^ thing it sees (that's a model) as the return. What use case has multiple models where we want the last one? Or I guess the other use case is we have stuff that ends in .joblib that aren't models?

Reply:
Yeah in this particular case the first file in the model zip is a label encoder, which can't be used to calculate shapley values . So the idea is that that model zip may contain multiple objects as joblib files, but not all of them are "models". We need the actual XGBModel to extract the shapley values.

So all this block does is go through each joblib file, load it, and check the type. If the type is XGBModel, it will return that. If not, it will continue to iterate through the list.

I guess the return statement could actually be in the "if" block to make it cleaner, then it would just return the first XGBModel object. But while there are multiple joblib files in the model zip, there should only be 1 joblib that produces an XGBModel object. As far as I know.

Reply:
To clarify, it doesn't always take the _last_ file. It takes the last file that is an XGBModel object. But that should only be one.